cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/validation/ValidationException.java

```diff
@@ -50,4 +54,25 @@ private static String generateMessage(List<ValidationFailure> failures) {
     return String.format("Errors were encountered during validation. %s",
         failures.isEmpty() ? "" : failures.iterator().next().getMessage());
   }
+
+  @Override
+  public String getFailureReason() {
+    return String.format("Stage '%s' encountered %s validation failures.", getFailureStage(),
+        failures.size());
+  }
+
+  @Override
+  public String getFailureStage() {
+    return !failures.isEmpty() ? failures.get(0).getStageName() : null;
```

Comment:
Mark it as `@Nullable` in the interface. It's actually good to mark all nullable in case some implementation can provide only partial information

Reply:
Added `default` implementations for `getErrorCategory`, `getErrorType` and made others `Nullable`.